internal/app/machined/pkg/runtime/v1alpha1/bootloader/grub/install.go

```diff
@@ -62,26 +66,63 @@ func (c *Config) Install(opts options.InstallOptions) (*options.InstallResult, e
 	return installResult, err
 }
 
-//nolint:gocyclo
+//nolint:gocyclo,cyclop
 func (c *Config) install(opts options.InstallOptions) (*options.InstallResult, error) {
 	if err := c.flip(); err != nil {
 		return nil, err
 	}
 
-	if err := utils.CopyFiles(
-		opts.Printf,
-		utils.SourceDestination(
-			opts.BootAssets.KernelPath,
-			filepath.Join(opts.MountPrefix, constants.BootMountPoint, string(c.Default), constants.KernelAsset),
-		),
-		utils.SourceDestination(
-			opts.BootAssets.InitramfsPath,
-			filepath.Join(opts.MountPrefix, constants.BootMountPoint, string(c.Default), constants.InitramfsAsset),
-		),
-	); err != nil {
-		return nil, err
+	if quirks.New(opts.Version).TalosVersionSDBootOnly() {
```

Comment:
I would probably instead check assets - if we see kernel & initramfs, use them, otherwise use UKI 

I mean check which file exists in the assets, it would be more compatible

Reply:
ah okay, makes things easier